synapse/storage/databases/main/event_push_actions.py

```diff
@@ -149,15 +154,15 @@ async def get_unread_event_push_actions_by_room_for_user(
 
     def _get_unread_counts_by_receipt_txn(
         self,
-        txn,
-        room_id,
-        user_id,
-        last_read_event_id,
-    ):
+        txn: LoggingTransaction,
+        room_id: str,
+        user_id: str,
+        last_read_event_id: Optional[str],
+    ) -> NotifCounts:
         stream_ordering = None
 
         if last_read_event_id is not None:
-            stream_ordering = self.get_stream_id_for_event_txn(
+            stream_ordering = self.get_stream_id_for_event_txn(  # type: ignore[attr-defined]
```

Comment:
I'm sure we could do something better here, but not sure it is worth blocking this PR for?